### PR TITLE
Feature/forms

### PR DESCRIPTION
**Pull request by olehhladkov**

### CODE REVIEW

File: src/api/deleteMovie.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { BASE_URL } from '../utils/constants';
+
+export default async (id) => {
```
Comment (Vladlan) [inline src/api/deleteMovie.js]: For development/debugging purposes it is better to have a name for a function (imagine you have code base without function names.

File: src/components/MoviesModal.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
     return null;
   }
 
   const isDeleteModal = type === 'delete';
 
+  const onMovieDelete = async (id) => {
+    await deleteMovie(id);
+    await dispatch(getMoviesThunk());
+
+    onClose();
```
Comment (Vladlan) [inline src/components/MoviesModal.js]: Add another argument in MoviesModal for onMovieDelete. Reason: onMovie close we should invoke onClose callback, on movie delete - onMovieDelete callback

File: src/components/MovieForm.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            value={formik.values.title}
+          />
+          {formik.touched.title && formik.errors.title ? (
+            <span className="error">{formik.errors.title}</span>
+          ) : null}
+        </div>
+        <div className="form-col">
+          <label htmlFor="release_date">release date</label>
+          <input
+            type="text"
```
Comment (Vladlan) [inline src/components/MovieForm.js]: For date use type "date" https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

File: src/components/MovieForm.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        <div className="form-col">
+          <label htmlFor="genres">genres</label>
+          <select
+            id="genres"
+            placeholder="Select genres"
+            multiple
+            onChange={formik.handleChange}
+            value={formik.values.genres}
+          >
+            {genreOptions.map((genre) => (
```
Comment (Vladlan) [inline src/components/MovieForm.js:133]: Did you try to make multiple selection with this?

**Review by Vladlan — CHANGES_REQUESTED** [review]
Screenshot 2023-03-27 at 11 58 14

Add form validation like on a MoviesAPI.  Make multiple selection and date input properly.

File: src/components/MovieForm.js (hunk 4 above)
Comment (olehhladkov) [reply src/components/MovieForm.js:133]: Yes, the screenshot is attached. 
Screenshot 2023-03-27 at 17 05 58

**Review by Vladlan — APPROVED** [review]
I hope you will not use such multiple select (as you've implemented it in this PR) at production projects, and even in your own projects. But requirements are met.

**Pull request merged**